api/hostdb_test.go

```
@@ -617,21 +618,25 @@ func TestHostDBAndRenterDownloadDynamicIPs(t *testing.T) {
 			t.Fatal(err)
 		}
 		// Give time for the upgrade to happen.
-		time.Sleep(time.Second * 3)
-	}
-
-	// Try downloading the file.
-	err = st.stdGetAPI("/renter/download/test?destination=" + downpath)
-	if err != nil {
-		t.Fatal(err)
-	}
-	// Check that the download has the right contents.
-	download, err = ioutil.ReadFile(downpath)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if bytes.Compare(orig, download) != 0 {
-		t.Fatal("data mismatch when downloading a file")
+		err = retry(100, time.Millisecond*100, func() error {
+			err = st.stdGetAPI("/renter/download/test?destination=" + downpath)
+			if err != nil {
+				return err
+			}
+			// Try downloading the file.
+			// Check that the download has the right contents.
+			download, err = ioutil.ReadFile(downpath)
+			if err != nil {
+				return err
+			}
+			if bytes.Compare(orig, download) != 0 {
+				return err
```

`err` could be nil here; need to return a new error. Also, use `bytes.Equal`